### PR TITLE
Add `serializerOpts` to server option interface

### PR DESCRIPTION
**Pull request by ddadaal**
`serializerOpts` is missing in the `FastifyServerOptions` interface. This PR adds it.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by RafaelGSS — APPROVED** [review]
LGTM.

File: fastify.d.ts [hunk 1]
```diff
@@ -102,6 +103,7 @@ export type FastifyServerOptions<
   onProtoPoisoning?: ProtoAction,
   onConstructorPoisoning?: ConstructorAction,
   logger?: boolean | FastifyLoggerOptions<RawServer> | Logger,
+  serializerOpts?: FJSOptions,
```
Comment (mcollina) [inline fastify.d.ts:106]: @Eomm could this be customized somehow?

**Review by mcollina — COMMENTED** [review]
Good work!

File: fastify.d.ts (hunk 1 above)
Comment (Eomm) [reply fastify.d.ts:106]: Yes, users who implemented a custom [`compilersFactory.buildSerializer`](https://www.fastify.io/docs/latest/Server/#schemacontroller) may have other options

Comment (mcollina) [reply fastify.d.ts:106]: How could we type that?

Comment (Eomm) [reply fastify.d.ts:106]: There is not a standard interface to apply. To avoid `any` we could set `Record<string, unknown>`

Comment (ddadaal) [reply fastify.d.ts]: Updated to `FJSOptions | Record<string, unknown>`

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**